internal/alloycli/cmd_run.go

```diff
@@ -192,6 +197,12 @@ func (fr *alloyRun) Run(configPath string) error {
 
 	level.Info(l).Log("boringcrypto enabled", boringcrypto.Enabled)
 
+	// Set the memory limit, this will honor GOMEMLIMIT if set
+	// If there is a cgroup will follow that
+	if fr.enableAutoMemoryLimit {
+		memlimit.SetGoMemLimitWithOpts(memlimit.WithLogger(slog.Default()))
+	}
+
```

Comment:
Rather than exposing this as a new configuration option, what do you think about having this automatically be enabled for users using experimental/public-preview, and then once we get enough information on it, we can decide whether it should always be on or needs to be configurable?

Reply:
IE if stability is experimental automatically enable? I can also set the flag as experimental, though unsure if we have a precedent for that with alloy.

Reply:
> IE if stability is experimental automatically enable? 

Yeah, like we would say that "auto memory limit is an experimental performance improvement" and only applies when `--stability.level=experimental. This is something I expect we'll be doing at some frequency, including for the eventual controller refactor.

> I can also set the flag as experimental

We don't have an existing precedent but I expect that most things will support feature gating at some point, including flags. 

However in this case I don't think marking the flag as experimental fully addresses my concerns, since presumably we intend for this to GA, and I prefer not adding too many more GA flags because of the increased API surface.

Reply:
Marked as public preview and added a helper function.